File: rb/features.rb
# ShikharMobile SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ShikharMobileFeatures
  def self.make_feature(name)
    case name
    when "base"
      ShikharMobileBaseFeature.new
    when "test"
      ShikharMobileTestFeature.new
    else
      ShikharMobileBaseFeature.new
    end
  end
end

File: py/features.py
# ShikharMobile SDK feature factory

from feature.base_feature import ShikharMobileBaseFeature
from feature.test_feature import ShikharMobileTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ShikharMobileBaseFeature(),
        "test": lambda: ShikharMobileTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
